docs/10-lifecycle.md

```diff
@@ -0,0 +1,81 @@
+# Lifecycle
```

Comment:
This is a great place to include a diagram of a lifecycle. 
![containerpilot-flow-withprestop](https://cloud.githubusercontent.com/assets/2766281/26504872/691610ae-4214-11e7-924e-fdbccb857661.jpg)